src/hotspot/share/utilities/bitMap.hpp

```diff
@@ -252,19 +254,38 @@ class BitMap {
   // Verify [beg,end) is a valid range, e.g. beg <= end <= size().
   void verify_range(idx_t beg, idx_t end) const NOT_DEBUG_RETURN;
 
-  // Iteration support.  Applies the closure to the index for each set bit,
-  // starting from the least index in the range to the greatest, in order.
-  // The iteration terminates if the closure returns false.  Returns true if
-  // the iteration completed, false if terminated early because the closure
-  // returned false.  If the closure modifies the bitmap, modifications to
-  // bits at indices greater than the current index will affect which further
-  // indices the closure will be applied to.
-  // precondition: beg and end form a valid range.
-  template <class BitMapClosureType>
-  bool iterate(BitMapClosureType* cl, idx_t beg, idx_t end);
-
-  template <class BitMapClosureType>
-  bool iterate(BitMapClosureType* cl);
+  // Applies an operation to the index for each set bit in [beg, end), in
+  // increasing order.  The iteration terminates if the operation returns
+  // false.
+  //
+  // If the operation modifies the bitmap, modifications to bits at indices
+  // greater than the current index will affect which further indices the
+  // operation will be applied to.
+  //
+  // precondition: beg and end form a valid range for the bitmap.
+  //
+  // If i is an index of the bitmap,
+  //
+  // - function(i) is a valid expression.  If the result is void then it is
+  // treated as if it always returned true, i.e. the iteration never
+  // terminates early.  Otherwise, the result must be convertible to bool.
+  //
+  // - cl->do_bit(i) is a valid expression whose result is convertible to bool.
```

Comment:
This description was an intermediate that I wasn't finished with, and isn't correct since it doesn't
describe `cl->do_bit(i)` as potentially returning void, which is supported by the implementation.
I seem to have lost the final version, and will need to rewrite it.